packages/shared/src/schemas.ts

```diff
@@ -10,12 +10,34 @@ export const quoteQuerySchema = z.object({
 
 export type QuoteQueryParams = z.infer<typeof quoteQuerySchema>;
 
+export type ByteString = string | `0x${string}`;
+export interface CcmMetadata {
+  gas_budget: string | number;
+  message: ByteString;
+  source_address: string;
+  source_chain: 'Bitcoin' | 'Ethereum' | 'Polkadot';
+  cf_parameters?: ByteString;
+}
+
+export const ccmMetadataSchema = z.object({
+  gas_budget: z.union([numericString, z.number()]),
+  message: z.union([hexString, z.string()]),
+  source_address: z.string(),
+  source_chain: z.union([
```

Comment:
we can infer this from the `srcAsset`

Reply:
might as well keep it so we can handle arbitrum in the future

Reply:
thats fine for me in general, but we should add it to the root of the `postSwapSchema`, bot inside the `ccmMetadata` then

Reply:
would also need to add `destChain` for the `destAsset` to keep things consistent

Reply:
i think we should change the shape of the root object when arbitrum is ready so we can capture all the changes together (broker api, front-end) etc.. Don't want to increase the scope of this ticket.  @acdibble @niklasnatter